README.md

```diff
@@ -70,7 +70,46 @@ When changes are complete, simply run `make generate-all` (`docker` currently re
 
 #### Other swagger definitions?
- - The goal of this repo is to not only provide a central, trusted location for consumers of our external APIs, but also to provide a re-usable means for service maintainers to easily maintain their service API definition. As this repository is more widely adopted, it is expected that we remove duplicate definitions from our organization's codebase where appropriate, rather than continue providing an API definition in multiple locations.
+ - The goal of this repo is to not only provide a central, trusted location for consumers of our external APIs, but also to provide a re-usable means for service maintainers to easily maintain their service API definition. **As this repository is more widely adopted, it is expected that we remove duplicate definitions from our organization's codebase where appropriate, rather than continue providing an API definition in multiple locations.**
+
+
+## How to use:
+
+#### Prologue:
+
+From swagger, client *libraries* can be generated. From client libraries, generated or not, client *applications* can be created. This repository, specifically the [`contracts`](./contracts/) directory, contains swagger definitions to generate client libraries. In go, there is a standard library that is made up of a plethora of smaller more granularly defined libraries. The contracts in this repo allow consumers to generate granular libraries so user's don't have to import an entire library, when they only care about a portion of the utility.
+
+It is intended that an individual client library is generated for each service defined. Cloud/oss definitions are the exception because we wanted to postpone splitting up those definitions into the bits that make them up in order to avoid breaking existing functionality. As this repository gains adoption, we would like to break subsets of the cloud/oss definitions into more granular ones - some examples of these granular services are query, write, and tasks.
+
+These more specialized swagger definitions (similar to the `datasourcesd`, `mapsd`, and `managed-functions` definitions), will allow the creation of granular libraries. The creation of more granular client libraries will allow developers to avoid bloat in their applications while still preserving the ability to consume a library to perform any action against our full api. Again, this workflow is much preferred to any developer over being mandated to consume a single client library that provides functionality to interact with an entire api surface are.
```

Comment:
```suggestion
These more specialized swagger definitions (similar to the `datasourcesd`, `mapsd`, and `managed-functions` definitions), will allow the creation of granular libraries. The creation of more granular client libraries will allow developers to avoid bloat in their applications while still preserving the ability to consume a library to perform any action against our full API. Again, this modular workflow is preferred and benefits developers by not mandating that they consume a single client library that supports the entire API surface area.
```